src/test/test_weighted_shuffle.cc

```diff
@@ -0,0 +1,63 @@
+// -*- mode:C++; tab-width:8; c-basic-offset:2; indent-tabs-mode:t -*-
+// vim: ts=8 sw=2 smarttab
+/*
+ * Ceph - scalable distributed file system
+ *
+ * This is free software; you can redistribute it and/or
```

Comment:
also, the major part of this file comes from https://wandbox.org/permlink/SuTYLHHG8brNNieq . probably we cannot claim it LGPL unless the author claims otherwise.